File: src/main/webapp/static/app/model/Person.js
Ext.define('phase3.model.Person', {
    extend: 'Ext.data.Model',
    alias: 'model.Person',
    idProperty:'id',
    fields: [
        {name: 'id', type: 'int'},
        {name: 'picture', type: 'string'},
        {name: 'firstName', type: 'string'},
        {name: 'lastName', type: 'string'},
        {name: 'gender', type: 'string'},
        {name: 'email', type: 'string'},
        {name: 'dateOfBirth', type: 'date'},
        {
            name: 'address',
            reference: 'phase3.model.Address'
        },
        {
            name: 'phones',
            reference: 'phase3.model.Phone'
        },
    ],
    validators: {
        gender: { type: 'inclusion', list: ['M', 'F'] },
        firstName: [
            {type: 'format', matcher: /^[A-Z][a-z]*(\\s(([a-z]{1,3})|(([a-z]+\\')?[A-Z][a-z]*)))*$/i},
            {type: 'length', min: 4, max: 10}
        ],
        lastName: [
            {type: 'format', matcher: /^[A-Z][a-z]*(\\s(([a-z]{1,3})|(([a-z]+\\')?[A-Z][a-z]*)))*$/i},
            {type: 'length', min: 5, max: 20}
        ]
    }
});

File: src/main/java/com/phase3/model/Person.java
package com.phase3.model;

import com.fasterxml.jackson.annotation.JsonFormat;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.hibernate.annotations.LazyCollection;
import org.hibernate.annotations.LazyCollectionOption;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import javax.persistence.*;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity(name="Person")
@Inheritance(strategy = InheritanceType.JOINED)
@Table(schema = "public", name = "persons")
public class Person {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id", unique=true, nullable = false)
  private long id;

  @Column(name = "first_name")
  private String firstName;

  @Column(name = "last_name")
  private String lastName;

  @Column(name = "gender")
  private char gender;

  @Column(name = "picture")
  private String picture;

  @Column(name = "email")
  private String email;

  @Column(name = "date_of_birth")
  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern="yyyy-MM-dd")
  private LocalDate dateOfBirth;

  @JsonIgnore
  @ManyToMany(fetch = FetchType.EAGER, cascade={CascadeType.PERSIST, CascadeType.MERGE})
  @LazyCollection(LazyCollectionOption.FALSE)
  @JoinTable(schema = "public", name = "persons_to_phones",
      joinColumns = @JoinColumn(name = "person_id"),
      inverseJoinColumns = @JoinColumn(name = "phone_id"))
  private List<Phone> phones = new ArrayList<>();

  @ManyToOne(fetch = FetchType.EAGER, cascade = {CascadeType.ALL})
  @JoinColumn(name = "address_id")
  private Address address;

  public Person() {
  }

  public Long getId() {
    return id;
  }

  public void setId(long id) {
    this.id = id;
  }

  public String getFirstName() {
    return firstName;
  }

  public void setFirstName(String firstName) {
    this.firstName = firstName;
  }

  public String getLastName() {
    return lastName;
  }

  public void setLastName(String lastName) {
    this.lastName = lastName;
  }

  public char getGender() {
    return gender;
  }

  public void setGender(char gender) {
    this.gender = gender;
  }

  public String getPicture() {
    return picture;
  }

  public void setPicture(String picture) {
    this.picture = picture;
  }

  public String getEmail() {
    return email;
  }

  public void setEmail(String email) {
    this.email = email;
  }

  public LocalDate getDateOfBirth() {
    return dateOfBirth;
  }

  public void setDateOfBirth(LocalDate dateOfBirth) {
    this.dateOfBirth = dateOfBirth;
  }

  public List<Phone> getPhones() {return phones;}

  public void setPhones(List<Phone> phones) { this.phones = phones; }

  public Address getAddress() {return address;}

  public void setAddress(Address address) {this.address = address;}

  public Person(long id, String firstName, String lastName, char gender, String picture, String email, LocalDate dateOfBirth, List<Phone> phones, Address address) {
    super();
    this.firstName = firstName;
    this.lastName = lastName;
    this.gender = gender;
    this.picture = picture;
    this.email = email;
    this.dateOfBirth = dateOfBirth;
    this.phones = phones;
    this.address = address;
  }

  @Override
  public int hashCode() {
    return getClass().hashCode();
  }

  @Override
  public boolean equals(Object obj) {
    if (this == obj) return true;
    if (!(obj instanceof Person)) return false;
    return id != 0 && id == (((Person) obj).getId());
  }

}
